app/src/main/java/com/example/cocktaildictionary/MainActivity.kt

```diff
@@ -34,18 +38,44 @@ class MainActivity : AppCompatActivity() {
             .subscribeOn(Schedulers.io())
             .subscribe(
                 { res ->
-                    binding.getData.visibility = View.INVISIBLE
-                    binding.progressBar.visibility = View.INVISIBLE
-                    binding.Id.visibility = View.VISIBLE
-                    binding.Id.text = "Success: Found " + res.Cocktails.size.toString() + " Cocktails"
+
+                    dataReceivedUITransition()
+                    cocktails = res.Cocktails
+                    binding.cocktailRecyclerview.setHasFixedSize(true)
+                    cocktailAdapter = CocktailAdapter(cocktails, this)
```

Comment:
Here

Reply:
Removed the intialization of the cocktails variable